src/matcha_ml/state/matcha_state.py

```diff
@@ -195,41 +213,68 @@ def _parse_terraform_output_resource_name(
             Returns:
                 Tuple[str, str, str]: the resource output for matcha.state.
             """
-            resource_type: Optional[str] = None
+            resource_type = None
 
             for key in RESOURCE_NAMES:
                 if key in output_name:
-                    resource_type = key
+                    resource_type = MatchaResource(key)
                     break
 
             if resource_type is None:
                 raise MatchaInputError(
                     f"A valid resource type for the output '{output_name}' does not exist."
                 )
 
-            flavor_and_resource_name = output_name[len(resource_type) + 1 :]
+            flavor_and_resource_name = output_name[len(resource_type.name) + 1 :]
 
             flavor, resource_name = flavor_and_resource_name.split("_", maxsplit=1)
             resource_name = resource_name.replace("_", "-")
-            resource_type = resource_type.replace("_", "-")
+            resource_type.name = resource_type.name.replace("_", "-")
 
             return resource_type, flavor, resource_name
 
-        state_outputs: Dict[str, Dict[str, str]] = defaultdict(dict)
+        matcha_state = MatchaState(components=[])
 
-        for output_name, properties in terraform_output.items():
+        for output_name, output_value in terraform_output.items():
             (
                 resource_type,
                 flavor,
                 resource_name,
             ) = _parse_terraform_output_resource_name(output_name)
-            state_outputs[resource_type].setdefault("flavor", flavor)
-            state_outputs[resource_type][resource_name] = properties["value"]  # type: ignore
+
+            component = matcha_state.get_component(resource_type.name)
+
+            if component:
```

Comment:
nit: might be a bit easier to read if this was an `if component is not None`

Reply:
Added, thanks.